src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.cs

```diff
@@ -830,10 +878,101 @@ internal static void EncryptKey(SafeKeyHandle safeKeyHandle, byte[] pbKey, int c
             Array.Reverse(pbEncryptedKey);
         }
 
+        internal static int EncryptData(
+            SafeKeyHandle hKey,
+            byte[] input,
+            int inputOffset,
+            int inputCount,
+            byte[] output,
+            int outputOffset,
+            int outputCount,
+            bool isFinal)
+        {
+            VerifyValidHandle(hKey);
+            Debug.Assert(input != null);
+            Debug.Assert(inputOffset >= 0);
+            Debug.Assert(inputCount >= 0);
+            Debug.Assert(inputCount <= input.Length - inputOffset);
+            Debug.Assert(output != null);
+            Debug.Assert(outputOffset >= 0);
+            Debug.Assert(outputCount >= 0);
+            Debug.Assert(outputCount <= output.Length - outputOffset);
+            Debug.Assert((inputCount % 8) == 0);
+            Debug.Assert((outputCount % 8) == 0);
+
+            // Figure out how big the encrypted data will be
+            // REVIEW NOTE: we could just bump up the buffer by 8 bytes always instead of doing the CryptEncrypt check for buffer size
```

Comment:
(And even then only when `isFinal` is true.  And even then only in CBC mode.  But it's definitely safest to just ask the crypto library how big it is)